rfqmsg/messages.go

```diff
@@ -115,6 +116,45 @@ func NewIncomingMsgFromWire(wireMsg WireMessage) (IncomingMsg, error) {
 	}
 }
 
+// WireMsgDataVersion specifies the version of the contents within a wire
+// message data field.
+type WireMsgDataVersion uint8
+
+const (
+	// V0 represents version 0 of the contents in a wire message data field.
+	V0 WireMsgDataVersion = 0
+)
+
+// WireMsgDataVersionEncoder is a function that can be used to encode a
+// WireMsgDataVersion to a writer.
+func WireMsgDataVersionEncoder(w io.Writer, val any, buf *[8]byte) error {
+	if version, ok := val.(*WireMsgDataVersion); ok {
+		versionUint8 := uint8(*version)
+		return tlv.EUint8(w, &versionUint8, buf)
```

Comment:
nit: why not `EUint8T` like in `asset.VersionEncoder`?

```go
return tlv.EUint8T(w, uint8(*version), buf)
```

Reply:
It should have been `tlv.EUint8T`. I'll try to keep that in mind in the future. I didn't get a chance to make that modification in this PR.

Reply:
Sg, could be in the follow-up PR you mentioned above.